### PR TITLE
[DSET-1856] Add single region deployment template.

### PR DESCRIPTION
**Pull request by ArthurKamalov**
This PR allows to run AWS `sam` CLI tool in order to deploy DataSet Cloudwatch integration for a particular region.

Since we publish our current integration by doing a very specific things to our template, it's very hard to re-use it for something else, so there was no other option than creating a new template.

### CODE REVIEW

File: cloudwatch_logs/single_region/template.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+
+Description: >-
+  Scalyr CloudWatch Logs Importer
+
+Parameters:
+  BaseUrl:
+    Description: Base URL of the Scalyr API
+    Type: String
```
Comment (jmakar-scalyr) [inline cloudwatch_logs/single_region/template.yml:10]: Should add https://app.scalyr.com as a default?

**Review by jmakar-scalyr — APPROVED** [review]
LGTM (though I am not familiar with CloudFormation)

File: cloudwatch_logs/single_region/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Please refer to our [docs](https://app.scalyr.com/solutions/import-cloudwatch) for the general information, 
+but keep in mind that we do not use template and functions that are offered by the docs and will build and deploy our 
+own artifacts. 
+
+In order to deploy this integration, please follow the next steps:
+
+1. Build lambda functions from sam template:
+    ```bash
+    sam build --template-file single_region/template.yml
```
Comment (dparker-scalyr) [inline cloudwatch_logs/single_region/README.md:11]: Where does the `sam` tool come from? It's not part of the awscli, so the dependency needs to be called out in either a prerequisites step or step 1.

File: cloudwatch_logs/single_region/template.yml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  KMSDecryptPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'kms:Decrypt'
+            Resource: !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*'
```
Comment (dparker-scalyr) [inline cloudwatch_logs/single_region/template.yml:121]: This won't work in govcloud, as the arn naming convention is different in govcloud. https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-arns.html

File: cloudwatch_logs/single_region/README.md (hunk 2 above)
Comment (ArthurKamalov) [reply cloudwatch_logs/single_region/README.md:16]: Added the Prerequisites step. Thanks.

File: cloudwatch_logs/single_region/template.yml (hunk 3 above)
Comment (ArthurKamalov) [reply cloudwatch_logs/single_region/template.yml]: Ah, sure. It completely slipped my mind, sorry. 
Tsis commit https://github.com/scalyr/scalyr-aws-serverless/commit/5658d9368c97305ae7e6792cab02eb3b7f4d7070 has to add this.

File: cloudwatch_logs/single_region/template.yml (hunk 1 above)
Comment (ArthurKamalov) [reply cloudwatch_logs/single_region/template.yml:10]: I heard that there has to be another endpoint, so for now I just decided to not mix them.

File: cloudwatch_logs/single_region/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      --s3-prefix <bucket_prefix> \
+      --parameter-overrides "${TEMPLATE_PARAMS}" \
+      --capabilities CAPABILITY_IAM
+    ``` 
+
+     Where:
+      * ``--stack-name`` Name of the CloudFormation stack.
+      * ``--region`` Name of a target region.
+      * ``--s3-bucket`` Name of the bucket to save deployment packages of the deployed lambda functions. 
+         Bucket has to be already created. Alternatively, the ``--resolve-s3`` option can be used to create bucket "of the fly" by the ``sam``
```
Comment (dkoshe) [inline cloudwatch_logs/single_region/README.md:49]: Typo -- not critical.
`of the fly` --> `on the fly`

Comment (ArthurKamalov) [reply cloudwatch_logs/single_region/README.md:49]: Oops, thanks.

**Review by dparker-scalyr — APPROVED** [review]

**Pull request merged**